src/smc-webapp/project/ask-filename.tsx

```diff
@@ -0,0 +1,106 @@
+import { Component, React, Rendered } from "../app-framework";
+const { default_filename } = require("../account.coffee");
+const { file_options } = require("../editor");
+const {
+  Col,
+  Row,
+  ButtonToolbar,
+  ControlLabel,
+  Button,
+  Form
+} = require("react-bootstrap");
+const { SearchInput } = require("../r_misc");
+const { IS_TOUCH } = require("../feature");
+
+interface Props {
+  actions: any;
+  ext_selection: string;
+  current_path: string;
+}
+interface State {
+  new_name: string;
+}
+
+export class AskNewFilename extends Component<Props, State> {
+  displayName = "ProjectFiles-AskNewFilename";
+  constructor(props) {
+    super(props);
+    this.state = {
+      new_name: default_filename()
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ new_name: default_filename() });
```

Comment:
Why is this called in the constructor and in `didMount`?

Reply:
It should probably only be called in componentDidMount -- it needs to get updated since it is time based.